internal/authz/authz.go

```diff
@@ -0,0 +1,237 @@
+func (a *Authorizer) AuthorizeCreateStore(ctx context.Context, clientID string) error {
+	relation, err := a.getRelation(CreateStore)
+	if err != nil {
+		return err
+	}
+	authorized, err := a.individualAuthorize(ctx, clientID, relation, a.getSystem(), &openfgav1.ContextualTupleKeys{})
+	if err != nil {
+		return err
+	}
+
+	if !authorized {
+		return status.Error(codes.Code(ErrorResponse.GetCode()), ErrorResponse.GetMessage())
+	}
+
+	return nil
+}
+
+// AuthorizeListStores checks if the user has access to list stores.
+func (a *Authorizer) AuthorizeListStores(ctx context.Context, clientID string) error {
+	relation, err := a.getRelation(ListStores)
+	if err != nil {
+		return err
+	}
+
+	authorized, err := a.individualAuthorize(ctx, clientID, relation, a.getSystem(), &openfgav1.ContextualTupleKeys{})
+	if err != nil {
+		return err
+	}
+	if !authorized {
+		return status.Error(codes.Code(ErrorResponse.GetCode()), ErrorResponse.GetMessage())
+	}
+
+	return nil
+}
+
+// ListAuthorizedStores returns the list of stores that the user has access to.
+func (a *Authorizer) ListAuthorizedStores(ctx context.Context, clientID string) ([]string, error) {
+	err := a.AuthorizeListStores(ctx, clientID)
+	if err != nil {
+		return nil, err
+	}
+
+	req := &openfgav1.ListObjectsRequest{
+		StoreId:              a.config.StoreID,
+		AuthorizationModelId: a.config.ModelID,
+		User:                 a.getApplication(clientID),
+		Relation:             CanCallGetStore,
+		Type:                 StoreType,
+	}
+
+	// Disable authz check for the list objects request.
+	ctx = authcontext.ContextWithSkipAuthzCheck(ctx, true)
+	resp, err := a.server.ListObjects(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	authcontext.ContextWithSkipAuthzCheck(ctx, false)
+
+	return resp.GetObjects(), nil
+}
+
+func (a *Authorizer) individualAuthorize(ctx context.Context, clientID, relation, object string, contextualTuples *openfgav1.ContextualTupleKeys) (bool, error) {
+	req := &openfgav1.CheckRequest{
+		StoreId:              a.config.StoreID,
+		AuthorizationModelId: a.config.ModelID,
+		TupleKey: &openfgav1.CheckRequestTupleKey{
+			User:     a.getApplication(clientID),
+			Relation: relation,
+			Object:   object,
+		},
+		ContextualTuples: contextualTuples,
+	}
+
+	// Disable authz check for the check request.
+	ctx = authcontext.ContextWithSkipAuthzCheck(ctx, true)
+	resp, err := a.server.Check(ctx, req)
+	if err != nil {
+		return false, err
+	}
+	authcontext.ContextWithSkipAuthzCheck(ctx, false)
+
+	if !resp.GetAllowed() {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (a *Authorizer) getStore(storeID string) string {
+	return fmt.Sprintf(`%s:%s`, StoreType, storeID)
+}
+
+func (a *Authorizer) getApplication(clientID string) string {
+	return fmt.Sprintf(`%s:%s`, ApplicationType, clientID)
+}
+
+func (a *Authorizer) getSystem() string {
+	return fmt.Sprintf(`%s:%s`, SystemType, RootSystemID)
+}
```

Comment:
These functions do not need to be methods of `Authorizer` as they do not depend on the `Authorizer` object in any way.

Also, another approach that lends itself to the type system:
```go
type StoreId string

func (s StoreId) String() string {
	return fmt.Sprintf("%s:%s", StoreType, string(s))
}

type ClientId string

func (c ClientId) String() string {
	return fmt.Sprintf("%s:%s", ApplicationType, string(c))
}

// assuming `SystemType` and `RootSystemID` are intended to be constants
var System = fmt.Sprintf("%s:%s", SystemType, RootSystemID)
```
Can be used as:
```go
storeId := StoreId("some_identifier")
output := fmt.Sprintf("%s", storeId) // output == "store:some_identifier"
```